google-login-plugin/src/com/google/gct/login/stats/UsageTrackerManager.java

```diff
@@ -0,0 +1,50 @@
+package com.google.gct.login.stats;
+
+import com.intellij.ide.util.PropertiesComponent;
+import com.intellij.openapi.diagnostic.Logger;
+
+/**
+ * Stores the users choice to opt in/out of sending usage metrics via the Google Usage Tracker.
+ */
+public class UsageTrackerManager {
+    private static final Logger LOG = Logger.getInstance(UsageTrackerManager.class);
+    private static final String USAGE_TRACKER_KEY = "GOOGLE_CLOUD_TOOLS_USAGE_TRACKER_OPT_IN";
+    private static final String USAGE_TRACKER_PROPERTY = "usage.tracker.property";
+
+    private UsageTrackerManager() {}
+
+    public static void setOptIn(boolean optIn) {
+        PropertiesComponent.getInstance().setValue(USAGE_TRACKER_KEY, String.valueOf(optIn));
+    }
+
+    public static boolean getOptIn() {
+        return PropertiesComponent.getInstance().getBoolean(USAGE_TRACKER_KEY, false);
+    }
+
+    /**
+     * @return true if the user has opted in/out of usage tracking; false otherwise
+     */
+    public static boolean isTrackingConfiguredByUser() {
+        return PropertiesComponent.getInstance().getValue(USAGE_TRACKER_KEY) != null;
+    }
+
+    /**
+     * @return true if the usage tracking property exists; false otherwise
+     */
+    public static boolean isTrackingConfigured() {
```

Comment:
That's a good point.  I think a clearer UsageTrackerManager API might be something super pedantic/explicit like:

// true if the user has agreed or disagreed to usage tracking
boolean userHasRecordedTrackingPreference()

void allowUsageTrackingForUser()
void disallowUsageTrackingForUser()

// true if the user has explicitly given consent false otherwise
boolean  hasUserAllowedTracking()

//potentially the above methods could all throw an IllegalStateException if isUsageTrackingAvailable() == false.

// true if we're on the right platform and have an analytics ID.
boolean isUsageTrackingAvailable()

Let me know what you think.